src/DynamoCoreWpf/ViewModels/Core/Converters/SerializationConverters.cs

```diff
@@ -107,13 +109,28 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
             writer.WriteValue(anno.GUID.ToString("N"));
             writer.WritePropertyName("Title");
             writer.WriteValue(anno.AnnotationText);
+            writer.WritePropertyName("DescriptionText");
+            writer.WriteValue(anno.AnnotationDescriptionText);
+            writer.WritePropertyName(nameof(ExtraAnnotationViewInfo.IsExpanded));
+            writer.WriteValue(anno.IsExpanded);
+            writer.WritePropertyName(nameof(ExtraAnnotationViewInfo.WidthAdjustment));
+            writer.WriteValue(anno.WidthAdjustment);
+            writer.WritePropertyName(nameof(ExtraAnnotationViewInfo.HeightAdjustment));
+            writer.WriteValue(anno.HeightAdjustment);
             writer.WritePropertyName("Nodes");
             writer.WriteStartArray();
             foreach (var m in anno.Nodes)
             {
                 writer.WriteValue(m.GUID.ToString("N"));
             }
             writer.WriteEndArray();
+            writer.WritePropertyName("Groups");
```

Comment:
@mjkkirschner @aparajit-pratap @pinzart @sm6srw What are your thoughts on this? I feel we are no longer strict on CoGS format so this could be fine? Or we can maybe add the nesting info to view layer but that may cause deserialization problem? I am not sure.

Reply:
I think this is fine, it makes sense to add this serialization in the annotation, but aren't we going to end up serializing these ids twice ⚠️ ?
Once in the nodes property and once in the groups property? Maybe we only need the NodesProperty and it really means "models"?

Reply:
Yeah we already have a `Annotation` block in view, originally I was thinking if we can just leverage that block. Here is an example:
```
 "Annotations": [
      {
        "Id": "9150975dc1b045ab9214c57533c519f3",
        "Title": "Test Group",
        "Nodes": [
          "5b151b86d7c04a2a9874fda51ae526c3",
          "6434e70dd8004aca9c26cbcf1a3dca64",
          "d58a319a377d40a3b80ffefbaaaff267",
          "621a399415ba495d9924ee39138bfb74"
        ],
        "Left": 181.19999999999987,
        "Top": 48.40000000000002,
        "Width": 578.40000000000009,
        "Height": 316.2,
        "FontSize": 36.0,
        "InitialTop": 101.60000000000002,
        "InitialHeight": 293.0,
        "TextblockHeight": 43.2,
        "Background": "#FFC1D676"
      }
    ]
```

Reply:
yea good points.. i originally added the `Groups` prop because i was running into issues when deserializing annotationModels that had nested groups. The issue being that we need to create any AnnotationModels that belong to other groups first, otherwise we cannot add them to the group we are creating.

I changed this now so that we serialize the `HasNestedGroups` property, and when deserializing i use a queue to build all nested groups first
```csharp
private void LoadAnnotations(IEnumerable<ExtraAnnotationViewInfo> annotationViews)
{
    if (annotationViews == null) return;

    var annotationQueue = new Queue<ExtraAnnotationViewInfo>(annotationViews);
    while (annotationQueue.Any())
    {
        var annotationViewInfo = annotationQueue.Dequeue();
        // Before creating this group we need to create
        // any group belonging to this group.
        if (annotationViewInfo.HasNestedGroups &&
            !annotationQueue.All(x=>x.HasNestedGroups))
        {
            annotationQueue.Enqueue(annotationViewInfo);
            continue;
        }

        LoadAnnotation(annotationViewInfo);
    }
}
```
https://github.com/DynamoDS/Dynamo/pull/11912/files#diff-fa18da253bafe5cc21047083856d6bc7b2d2b7f49d1441c589b274d405b947edR2172

Reply:
```json
{
  "Id": "7fbc02d54b664e38b37a639e6c9d9663",
  "Title": "GroupWithGroupedGroup",
  "DescriptionText": "<Click here to edit the group description>",
  "IsExpanded": true,
  "WidthAdjustment": 0.0,
  "HeightAdjustment": 0.0,
  "Nodes": [
    "1ffbaf75bf12424b8f6f55fdcea208dc",
    "3a8039e80d984ae19f1a4dfd08a49289",
    "1997f3c83f764260a3f2f10cc73f802d"
  ],
  "HasNestedGroups": true,
  "Left": 185.70075726305083,
  "Top": 1104.2283106882585,
  "Width": 623.505028610443,
  "Height": 392.58280439753753,
  "FontSize": 36.0,
  "InitialTop": 1177.5616440215917,
  "InitialHeight": 387.46579475350063,
  "TextblockHeight": 63.333333333333336,
  "Background": "#FFC1D676"
}